### PR TITLE
add golangci-lint support and drop Go 1.19

### PR DESCRIPTION
**Pull request by amery**

### CODE REVIEW

**Comment by amery** [comment]
meh, tests need refactoring

File: sync.go
Comment (karasz) [inline sync.go:1]: does this really belong in this PR?

**Review by karasz — APPROVED** [review]

File: sync.go
Comment (amery) [reply sync.go:1]: It was needed to make golangci-lint happy, but you are right, it should be before this PR

Comment (amery) [reply sync.go:1]: commits moved to https://github.com/darvaza-proxy/core/pull/42

**Comment by amery** [comment]
rebased after merging #42

**Review by karasz — APPROVED** [review]
+1

**Pull request merged**